### PR TITLE
Lava dwarfs 3: The testmerge era

### PR DESCRIPTION
**Pull request by ma44**
cool wiki for people that don't know what dorfs are
https://tgstation13.org/wiki/Lava_Dwarf
Hopefully feature freeze doesn't rekt this pr

### actually i may want this merged because goddamm all of these merge conflicts that will pop up

:cl: dwarfs or something
add: dorfs lavaland ruin, play as a dorf migrant
/:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 
Because dorfs was always fun, but it can still run even without GOAP additions, so I'm leaving it here so the playerbase can enjoy being dorfs if this gets testmerged for some reason

### CODE REVIEW

**Comment by ma44** [comment]
Oh neat conflicts

**Comment by RandomMarine** [comment]
Please.
Oh god please.
Disregard feature creep and just get this in a working, merge-able state.

**Comment by DoctoreMad** [comment]
Goof's PR has less conflicts than this one.

**Comment by ma44** [comment]
I'm not even going to understand how that works

File: code/game/turfs/simulated/minerals.dm [hunk 1]
```diff
@@ -395,14 +395,28 @@ GLOBAL_VAR_INIT(secret_triggered, FALSE)
...
+		visible_message("<span class = 'userdanger'>A tendril suddonly pops out of the ground!</span>")
+		if(prob(50))
+			tendril.mob_type = /mob/living/carbon/human/interactive/greytide/clown //c l o w n s
+			tendril.name = "Silly tendril" //I should really had made a subtype of tendrils instead of snowflaking vars
+		else
+			tendril.mob_type = /mob/living/simple_animal/hostile/swarmer/ai
+			tendril.name = "Strange tendril"
+			tendril.spawn_time = 150 //Half the time
+		if(prob(1))
+			GLOB.secret_triggered = FALSE //How to be unlucky
```
Comment (DoctoreMad) [inline code/game/turfs/simulated/minerals.dm]: Are you telling me there's a chance that two HFSs appear in the same round?

Comment (ma44) [inline code/game/turfs/simulated/minerals.dm]: a *chance*

**Comment by DoctoreMad** [comment]
RIP Lava Dwarves

**Comment by FantasticFwoosh** [comment]
Wait. Fuck. Already? Nice to see it back.

- GOAP if i remember correctly could mean we can have a dwarf fortress on lavaland that is 100% autonomous run by bots.

**Comment by ma44** [comment]
huh, turns out that i did the wrong version, hold on kids

File: code/game/turfs/simulated/minerals.dm (hunk 1 above)
Comment (ma44) [inline code/game/turfs/simulated/minerals.dm]: this is also outdated and not included, go figure

**Comment by Armhulen** [comment]
IT'S DEAD

**Comment by ma44** [comment]
Can't be ded if there's a ma44 on it or something

**Comment by ma44** [comment]
>realizing you can't edit the files because you made the pr of goofs branch which you don't have control over but since you use web editor you can't make a fork of that branch or something and stuff

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -286,6 +286,21 @@
...
+	name = "Wooden bucket"
+	desc = "It's a wooden bucket, don't question how it carries very hot liquids."
+	icon = 'icons/obj/janitor.dmi' //TODO: add the cool bucket sprite
+	icon_state = "bucket"
+	item_state = "bucket"
+	materials = list(MAT_METAL=200)
+	w_class = WEIGHT_CLASS_NORMAL
+	amount_per_transfer_from_this = 25 //Mods require 25u of reagents, saves a click
+	possible_transfer_amounts = list(10,15,20,25,30,50,70)
+	volume = 75 //Code adds 75u of reagents while the original bucket had 70u max, making it loose	flags = OPENCONTAINER
```
Comment (ma44) [inline code/modules/reagents/reagent_containers/glass.dm]: cool reminder for me to FIX THISSSSSSSSSSSSS

**Comment by DoctoreMad** [comment]
>GOAP if i remember correctly could mean we can have a dwarf fortress on lavaland that is 100% autonomous run by bots.

Yeah but why? For what purpose? I thought this was supposed to be for players.

**Comment by ma44** [comment]
For autistic forts and to make you be frustrated so you can go and scream at the computer about why your 2d spessmen lure a colossus into the base

**Comment by ma44** [comment]
# cant wake up

**Comment by AnonymousNow** [comment]
This sounds fun.

File: code/_onclick/hud/hud.dm [hunk 3]
```diff
@@ -21,6 +21,8 @@
 	var/obj/screen/alien_plasma_display
 	var/obj/screen/alien_queen_finder
 
+	var/obj/screen/alcohol_amount
```
Comment (tortellinitony) [inline code/_onclick/hud/hud.dm]: Unused var

File: code/_onclick/hud/hud.dm [hunk 4]
```diff
@@ -112,6 +114,7 @@
 	blobpwrdisplay = null
 	alien_plasma_display = null
 	alien_queen_finder = null
+	alcohol_amount = null
```
Comment (tortellinitony) [inline code/_onclick/hud/hud.dm]: Unused var

File: code/modules/mining/smelter.dm [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+		return
+	if(!istype(W, /obj/item/weapon/smith_hammer))
+		..()
+	if(!current_mold && istype(W, /obj/item/weapon/reagent_containers/glass/mold))
+		var/obj/item/weapon/reagent_containers/glass/mold/M = W
+		var/datum/reagent/R = M.reagents.get_master_reagent()
+		if(R && R.volume == 25)
+			if(user.drop_item())
+				to_chat(user, "You place [M] on [src].")
+				M.loc = src
```
Comment (tortellinitony) [inline code/modules/mining/smelter.dm]: `M.forceMove(src)`

File: code/modules/mob/living/carbon/human/interactive.dm [hunk 6]
```diff
@@ -83,7 +83,7 @@
 	var/traitorType = 0
 
 	var/voice_saved = FALSE
-
+	var/allow_random = 1
```
Comment (tortellinitony) [inline code/modules/mob/living/carbon/human/interactive.dm]: Use TRUE

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 7]
```diff
@@ -79,14 +79,19 @@
...
 /datum/chemical_reaction/soapification/on_reaction(datum/reagents/holder, created_volume)
 	var/location = get_turf(holder.my_atom)
 	for(var/i = 1, i <= created_volume, i++)
 		new /obj/item/weapon/soap/homemade(location)
 
+/datum/chemical_reaction/soapification/two
+	name = "Soapification2"
+	id = "soapification2"
+	required_reagents = list("ash" = 10, "water"  = 10)
+	mob_react = 1
```
Comment (tortellinitony) [inline code/modules/reagents/chemistry/recipes/others.dm]: Use TRUE

File: code/modules/mob/living/carbon/human/species_types/dwarf.dm [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+	var/mob/living/carbon/human/H = C
+	H.grant_language(/datum/language/dwarvish)
+	H.remove_language(/datum/language/common)
+	H.facial_hair_style = dwarf_hair
+	H.update_hair()
+
+	var/obj/item/organ/alcoholvessel/dwarf
+	dwarf = H.getorganslot("dwarf_organ")
+	if(!dwarf)
+		dwarf = new()
```
Comment (tortellinitony) [inline code/modules/mob/living/carbon/human/species_types/dwarf.dm]: Remove `()`

File: code/modules/surgery/organs/dwarf_organ.dm [hunk 9]
```diff
@@ -0,0 +1,85 @@
+/obj/item/organ/alcoholvessel //essentially the opposite of the xeno's plasmavessel, but with alcohol
+	name = "adamantine infused lungs"
+	icon_state = "plasma"
+	origin_tech = "biotech=5"
+	w_class = WEIGHT_CLASS_NORMAL
+	zone = "chest"
+	slot = "dwarf_organ"
```
Comment (tortellinitony) [inline code/modules/surgery/organs/dwarf_organ.dm]: Make an organ define DWARF_ORGAN

**Review by tortellinitony — CHANGES_REQUESTED** [review]

**Comment by ma44** [comment]
waitin years for pr to be merged at goofs branch

**Comment by Grazyn** [comment]
>GOAP if i remember correctly could mean we can have a dwarf fortress on lavaland that is 100% autonomous run by bots.

Would it be possible to give a ghost the ability to place structures and give simple orders to the dwarf bots? You know, so we could play ACTUAL dwarf fortress.

**Comment by LarissaMayrink** [comment]
>Would it be possible to give a ghost the ability to place structures and give simple orders to the dwarf bots? You know, so we could play ACTUAL dwarf fortress.
lava carps when

**Comment by FantasticFwoosh** [comment]
Whoever controls the sceptre controls the dwarves @Grazyn i guess, also @DoctoreMad its a fun simulation thing like swarmers on lavaland (granted that's not as much fun) and you could roll back the ghost roles to being dwarven nobles in charge of dwarf-bots as peasantry.

**Comment by Jalleo** [comment]
Well there are a few merge conflicts.

**Comment by ma44** [comment]
# Rip
https://github.com/Iamgoofball/-tg-station/pull/13#issuecomment-317619212

**Comment by kevinz000** [comment]
it's another ma44 episode
Change the channel.

**Comment by Tacolizard** [comment]
>web editor

**Pull request closed without merging**